### PR TITLE
Runner & Executor - Support AMD GPU

### PR DESCRIPTION
**Pull request by abuccts**
__Description__
Support both NVIDIA and AMD GPU and check GPU vendor during deployment and execution.

__Major Revisions__
* Add GPU environment check in `sb deploy`
* Check GPU vendor in executor.

### CODE REVIEW

File: superbench/runner/playbooks/deploy.yaml [hunk 1]
```diff
@@ -25,16 +25,26 @@
...
     - name: Set GPU Facts
       set_fact:
-        nvidia_gpu_exist: "{{ nvidia_dev.stat.ischr is defined and nvidia_dev.stat.ischr }}"
-        amd_gpu_exist: "{{ amd_dev.stat.ischr is defined and amd_dev.stat.ischr }}"
+        nvidia_gpu_exist: >-
+          {{ nvidia_dev.results[0].stat.ischr is defined and nvidia_dev.results[0].stat.ischr and
+          nvidia_dev.results[1].stat.ischr is defined and nvidia_dev.results[1].stat.ischr }}
+        amd_gpu_exist: >-
+          {{ amd_dev.results[0].stat.ischr is defined and amd_dev.results[0].stat.ischr and
+          amd_dev.results[1].stat.isdir is defined and amd_dev.results[1].stat.isdir }}
```
Comment (cp5555) [inline superbench/runner/playbooks/deploy.yaml:47]: Would you please write some comments to explain it?

File: superbench/runner/playbooks/deploy.yaml [hunk 2]
```diff
@@ -25,16 +25,26 @@
   tasks:
     - name: Checking NVIDIA GPU
```
Comment (cp5555) [inline superbench/runner/playbooks/deploy.yaml:26]: It's not check NVIDIA GPU, right? If you would like to check, you should use lspci.

File: superbench/common/devices/__init__.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""SuperBench devices module."""
+
+from superbench.common.devices.gpu import GPU
+
+__all__ = [
+    'GPU',
+]
```
Comment (cp5555) [inline superbench/common/devices/__init__.py:10]: Why do we put \_\_init\_\_.py in common directly. Currently, we reduce it from `superbench.common.devices.gpu.GPU` to `superbench.common.devices.GPU`

Comment (cp5555) [reply superbench/common/devices/__init__.py:10]: Besides, why not put into `utils`?

**Comment by cp5555** [comment]
Please add `__Major Revision__`

File: superbench/common/devices/gpu.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    def __init__(self):
+        """Initilize."""
+        self.vendor = self.get_vendor()
+        # TODO: check CUDA or ROCm availability accordingly
+        if self.vendor == 'nvidia':
+            self.platform = Platform.CUDA
+        elif self.vendor == 'amd':
+            self.platform == Platform.ROCM
+        else:
+            self.platform = Platform.CPU
```
Comment (guoshzhao) [inline superbench/common/devices/gpu.py:23]: `self.vendor` => `vendor`
`self.platform` => `self.__platform`
Then using @property to get platform?
```
@property
def platform(self):
    return self.__platform
```

File: superbench/executor/executor.py [hunk 5]
```diff
@@ -67,8 +68,7 @@ def __get_enabled_benchmarks(self):
 
     def __get_platform(self):
         """Detect runninng platform by environment."""
-        # TODO: check devices and env vars
-        return Platform.CUDA
+        return GPU().platform
```
Comment (guoshzhao) [inline superbench/executor/executor.py:71]: If we use the structure:
-devices
   --gpu
   --others
how __get_platform() to know devices? Here looks like a hardcode.

File: superbench/runner/playbooks/deploy.yaml (hunk 2 above)
Comment (abuccts) [reply superbench/runner/playbooks/deploy.yaml]: renamed

File: superbench/common/devices/gpu.py (hunk 4 above)
Comment (abuccts) [reply superbench/common/devices/gpu.py]: updated

File: superbench/executor/executor.py (hunk 5 above)
Comment (abuccts) [reply superbench/executor/executor.py]: updated

**Review by cp5555 — APPROVED** [review]

**Review by guoshzhao — APPROVED** [review]

**Pull request merged**